agent/consul/gateways/datastore/datastore.go

```diff
@@ -0,0 +1,15 @@
+package datastore
```

Comment:
This feels like we're creating a package for a single interface, and then we wind up stuttering `datastore.DataStore`. Is this beneficial over just putting the interface in the parent `gateways` package?

Reply:
I think at the time I had been worried about potential circular dependencies, but it ended up not being an issue. I can pull it into the gateways package.